packages/http-client/src/index.ts

```diff
@@ -722,6 +722,12 @@ export class HttpClient {
     }
 
     const usingSsl = parsedUrl.protocol === 'https:'
+
+    // Lazy load ProxyAgent to avoid bundling all the undici
+    const ProxyAgent =
+      // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+      require('undici/lib/proxy-agent') as typeof import('undici/types/proxy-agent').default
```

Comment:
This can be further improved by using dynamic imports. However, I didn't include it in this pull request, because using dynamic imports needs making `getAgentDispatcher` async, which changes the return type of a public method. Based on my research, this is only used in the github package, so it should be fine, but if we want to go with that, we need to have a major version bump for http-client. 
https://github.com/aminya/toolkit/commit/b716d6f9619c1b31f415ccd849abe1374f20a51c